connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -0,0 +1,226 @@
+        } catch (Exception e) {
+            log.error("Could not stop kafka", e);
+        }
+    }
+
+    public void startConnect() {
+        log.info("Starting standalone connect cluster..");
+        workerProps.put("bootstrap.servers", kafka().bootstrapServers());
+        workerProps.put("rest.host.name", "localhost");
+
+        putIfAbsent(workerProps, "group.id", "connect-integration-test-" + clusterName);
+        putIfAbsent(workerProps, "offset.storage.topic", "connect-offset-topic-" + clusterName);
+        putIfAbsent(workerProps, "offset.storage.replication.factor", "1");
+        putIfAbsent(workerProps, "config.storage.topic", "connect-config-topic-" + clusterName);
+        putIfAbsent(workerProps, "config.storage.replication.factor", "1");
+        putIfAbsent(workerProps, "status.storage.topic", "connect-storage-topic-" + clusterName);
+        putIfAbsent(workerProps, "status.storage.replication.factor", "1");
+        putIfAbsent(workerProps, "key.converter", "org.apache.kafka.connect.json.JsonConverter");
+        putIfAbsent(workerProps, "value.converter", "org.apache.kafka.connect.json.JsonConverter");
+        putIfAbsent(workerProps, "key.converter.schemas.enable", "false");
+        putIfAbsent(workerProps, "value.converter.schemas.enable", "false");
+
+        log.info("Scanning for plugins...");
+        Plugins plugins = new Plugins(workerProps);
+        plugins.compareAndSwapWithDelegatingLoader();
+
+        DistributedConfig config = new DistributedConfig(workerProps);
+
+        RestServer rest = new RestServer(config);
+        advertisedUrl = rest.advertisedUrl();
+        String workerId = advertisedUrl.getHost() + ":" + advertisedUrl.getPort();
+
+        KafkaOffsetBackingStore offsetBackingStore = new KafkaOffsetBackingStore();
+        offsetBackingStore.configure(config);
+
+        Worker worker = new Worker(workerId, kafkaCluster.time(), plugins, config, offsetBackingStore);
+
+        WorkerConfigTransformer configTransformer = worker.configTransformer();
+        Converter internalValueConverter = worker.getInternalValueConverter();
+
+        StatusBackingStore statusBackingStore = new KafkaStatusBackingStore(kafkaCluster.time(), internalValueConverter);
+        statusBackingStore.configure(config);
+
+        ConfigBackingStore configBackingStore = new KafkaConfigBackingStore(
+                internalValueConverter,
+                config,
+                configTransformer);
+
+        Herder herder = new DistributedHerder(config, kafkaCluster.time(), worker,
+                ConnectUtils.lookupKafkaClusterId(config), statusBackingStore, configBackingStore,
+                advertisedUrl.toString());
+        connect = new Connect(herder, rest);
+        connect.start();
+    }
+
+    public void startConnector(String connName, Map<String, String> connConfig) throws IOException {
+        String url = String.format("http://localhost:8083/connectors/%s/config", connName);
```

Comment:
I'd declare host and port somewhere, even though they are not meant to be configurable yet (basically the port)